packages/backend/src/outgoing_payment/service.ts

```diff
@@ -154,32 +154,79 @@ async function createOutgoingPayment(
 function requotePayment(
   deps: ServiceDependencies,
   id: string
-): Promise<OutgoingPayment> {
+): Promise<OutgoingPayment | OutgoingPaymentError> {
   return deps.knex.transaction(async (trx) => {
     const payment = await OutgoingPayment.query(trx).findById(id).forUpdate()
-    if (!payment) throw new Error('payment does not exist')
+    if (!payment) return OutgoingPaymentError.UnknownPayment
     if (payment.state !== PaymentState.Cancelled) {
-      throw new Error(`Cannot quote; payment is in state=${payment.state}`)
+      return OutgoingPaymentError.WrongState
     }
     await payment.$query(trx).patch({ state: PaymentState.Quoting })
     return payment
   })
 }
 
+export interface FundOutgoingPaymentOptions {
+  id: string
+  amount: bigint
+  transferId?: string
+}
+
+async function fundPayment(
+  deps: ServiceDependencies,
+  { id, amount, transferId }: FundOutgoingPaymentOptions
+): Promise<OutgoingPayment | OutgoingPaymentError> {
+  return deps.knex.transaction(async (trx) => {
+    const payment = await OutgoingPayment.query(trx)
+      .findById(id)
+      .forUpdate()
+      .withGraphFetched('account.asset')
```

Comment:
Can you double-check that the `forUpdate` doesn't also get carried over the the `account.asset` being fetched? (i.e. that the payment row is locked but not the account/asset). I know we do this elsewhere too but it didn't occur to me until now...

Reply:
https://github.com/Vincit/objection.js/issues/1608
I'm guessing it would for `withGraphJoined`